File: themes/lib/targets/zathura.py
"""Zathura PDF viewer color theme generator."""

from themes.lib.schema import ColorScheme, ThemeState

TARGET_NAME = "zathura"
ASSEMBLY = "import"
OUTPUT_PATH = "~/.config/zathura/colors"
RELOAD_CMD = None
COMMENT = "#"


def generate(colors: ColorScheme, state: ThemeState) -> str:
    return (
        f'set default-bg "{colors.bg}"\n'
        f'set default-fg "{colors.fg}"\n'
        f'set statusbar-bg "{colors.bg1}"\n'
        f'set statusbar-fg "{colors.fg}"\n'
        f'set inputbar-bg "{colors.bg}"\n'
        f'set inputbar-fg "{colors.fg}"\n'
        f'set notification-bg "{colors.bg1}"\n'
        f'set notification-fg "{colors.fg}"\n'
        f'set notification-error-bg "{colors.red}"\n'
        f'set notification-error-fg "{colors.fg}"\n'
        f'set notification-warning-bg "{colors.yellow}"\n'
        f'set notification-warning-fg "{colors.bg}"\n'
        f'set highlight-color "{colors.yellow}"\n'
        f'set highlight-active-color "{colors.accent}"\n'
        f'set completion-bg "{colors.bg1}"\n'
        f'set completion-fg "{colors.fg}"\n'
        f'set completion-highlight-bg "{colors.accent}"\n'
        f'set completion-highlight-fg "{colors.bg}"\n'
        f'set recolor-lightcolor "{colors.bg}"\n'
        f'set recolor-darkcolor "{colors.fg}"\n'
        f'set recolor "true"\n'
        f'set recolor-keephue "false"\n'
    )

File: desktopctl/src/theme/targets/zathura.rs
use super::{Assembly, GeneratedContent, TargetMetadata};
use crate::theme::schema::{ColorScheme, ThemeState};

pub const METADATA: TargetMetadata =
    TargetMetadata::new("zathura", Assembly::Import, &["color_scheme"])
        .output("~/.config/zathura/colors")
        .comment("#");

pub fn generate(colors: &ColorScheme, _state: &ThemeState) -> crate::Result<GeneratedContent> {
    Ok(GeneratedContent::text(format!(
        concat!(
            "set default-bg \"{}\"\n",
            "set default-fg \"{}\"\n",
            "set statusbar-bg \"{}\"\n",
            "set statusbar-fg \"{}\"\n",
            "set inputbar-bg \"{}\"\n",
            "set inputbar-fg \"{}\"\n",
            "set notification-bg \"{}\"\n",
            "set notification-fg \"{}\"\n",
            "set notification-error-bg \"{}\"\n",
            "set notification-error-fg \"{}\"\n",
            "set notification-warning-bg \"{}\"\n",
            "set notification-warning-fg \"{}\"\n",
            "set highlight-color \"{}\"\n",
            "set highlight-active-color \"{}\"\n",
            "set completion-bg \"{}\"\n",
            "set completion-fg \"{}\"\n",
            "set completion-highlight-bg \"{}\"\n",
            "set completion-highlight-fg \"{}\"\n",
            "set recolor-lightcolor \"{}\"\n",
            "set recolor-darkcolor \"{}\"\n",
            "set recolor \"true\"\n",
            "set recolor-keephue \"false\"\n",
        ),
        colors.bg,
        colors.fg,
        colors.bg1,
        colors.fg,
        colors.bg,
        colors.fg,
        colors.bg1,
        colors.fg,
        colors.red,
        colors.fg,
        colors.yellow,
        colors.bg,
        colors.yellow,
        colors.accent,
        colors.bg1,
        colors.fg,
        colors.accent,
        colors.bg,
        colors.bg,
        colors.fg,
    )))
}
